### PR TITLE
Add benchmark refactoring helper

### PR DESCRIPTION
**Pull request by jholton**
## Description of change
Adds a `benchmark` directory.
Contains a `runner.sh` file to assist with comparing benchmark results of refactored code.
The code you want to benchmark/compare goes in the `test_code.rb` file.

Example scenario:
You've been working on some improvements to EVSS Faraday middleware.
Inside `test_code` you put the following:

```
subject = EVSS::PCIU::Service.new FactoryBot.build(:user, :loa3)
Benchmark.ips do |bm|
  bm.report('evss_middleware') do
    VCR.use_cassette('evss/pciu/email') do
      subject.get_email_address
    end
  end
end
```

You compare it to a previous commit my running:
`./benchmark/runner.sh -b 8dfe3c5`

The runner will:
* checkout the given commit
* run the benchmarks
* checkout the branch you started on
* run the benchmarks

Results will look something like this:
```
`--> ./benchmark/runner.sh -b 8dfe3c5 -a 21d51bb
Warming up --------------------------------------
     evss_middleware    43.000  i/100ms
Calculating -------------------------------------
     evss_middleware    373.139  (±16.6%) i/s -      1.849k in   5.123281s
============================================================================================
Warming up --------------------------------------
     evss_middleware    42.000  i/100ms
Calculating -------------------------------------
     evss_middleware    446.112  (± 6.9%) i/s -      2.268k in   5.108511s
============================================================================================
Already on 'jh_benchmarking'
```

## Testing done
local

## Testing planned
N/A.  This is intended for development/testing scenarios only.

#### Unique to this PR

- [x] Adds a `benchmark` directory
- [x] Adds a script to assist in assessing benchmark improvements of refactored code.

#### Applies to all PRs

- [ ] ~~Appropriate logging~~
- [ ] ~~Swagger docs have been updated, if applicable~~
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] ~~Provide which alerts would indicate a problem with this functionality (if applicable)~~

### CODE REVIEW

**Comment by annaswims** [comment]
I'm not able to make th example work.  I don't see a commit `3cb6a926` in this repo.

When I try to run the example in `benchmark/test_code.rb`, I see:
 ````
Warming up --------------------------------------
     evss_middlewareW, [2019-07-22T19:41:34.615089 #5252]  WARN -- Betamocks: Mock response not found: [../vets-api-mockdata/evss/pciu/email/796111863.yml]
/Users/annacarey/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/bundler/gems/betamocks-dfeb140c9571/lib/betamocks/response_cache.rb:26:in `default_response': Betamocks default response requested but none exist. Please create one at: [../vets-api-mockdata/evss/pciu/email/default.yml]. (IOError)
 ````

Do you have different settings for betamocks?

**Comment by jholton** [comment]
`3cb6a926` was from one of my working branches.  I've updated the examples to use commits from `master`.

I'm surprised you haven't run into the `mockdata` issue before.  I took the example from `spec/lib/evss/pciu/service_spec.rb`, which uses the same factory user for the same EVSS call.

To quickly make it work, duplicate one of the existing files in `vets-api-mockdata/evss/pciu/email/` and rename it `default.yml`

Maybe we should make some issues (sprint objectives?) around making sure our mockdata lines up with what are specs are requesting.

But I digress.  All feedback welcome on this.  @kreek had mentioned refactoring faraday middlewares in the [original issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/131), and that got me brainstorming about how to measure the performance improvements of a refactoring, which led to this PR.

Is this a good idea overall?  Or unnecessary?
How is the directory structure?  Is it OK at root, or should it be nested somewhere (e.g. `spec`)?
I like it where it is.  Although they both help ensure code quality, `benchmarks` are not `specs`.  Plus having it in the root helps keep code performance more present in mind since you'll notice the `benchmark` directory more often.

I also don't intend this script to be the sole resident of `benchmark`.  As we need and created more scripted benchmarking tools, they can join.

File: benchmark/runner.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Runner script for comparing the performance of code changes
+# optional flags:
+#   -b  supply the "before" commit SHA. Defaults to HEAD~1
+#   -a  supply the "after" commit SHA.  Defaults to current branch
+#   -f  supply the file name that contains the code to benchmark. Defaults to 'benchmark/test_code.rb'
+#
+# Full example: ./benchmark/runner.sh -b 8dfe3c5 -a 21d51bb -f benchmark/testing.rb
```
Comment (annaswims) [inline benchmark/runner.sh]: nitpick: `test_code.rb`

Comment (jholton) [reply benchmark/runner.sh]: I was trying to show an instance where the user specifies a _different_ file since `test_code` is used by default.

**Pull request merged**